libc/src/math/generic/canonicalizef.cpp

```diff
@@ -0,0 +1,26 @@
+//===-- Implementation of canonicalizef function
+//----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------------------===//
+
+#include "src/math/canonicalizef.h"
+#include "src/__support/FPUtil/FEnvImpl.h"
+#include "src/__support/FPUtil/FPBits.h"
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(int, canonicalizef, (float *cx, const float *x)) {
+  using FPB = fputil::FPBits<float>;
```

Comment:
These implementations are identical.  You can refactor them to a template in https://github.com/llvm/llvm-project/blob/main/libc/src/__support/FPUtil/BasicOperations.h